solarforecastarbiter/reports/figures.py

```diff
@@ -46,7 +50,29 @@ def construct_fx_obs_cds(fx_values, obs_values):
         tz-aware input times are converted to tz-naive times in the
         input time zone.
     """
-    data = pd.DataFrame({'observation': obs_values, 'forecast': fx_values})
+    frames = []
+    for pfxobs in report.raw_report.processed_forecasts_observations:
+        frame_dict = {
+            'observation_values': pfxobs.observation_values,
+            'forecast_values': pfxobs.forecast_values,
+            'observation_name': _obs_name(pfxobs.original),
```

Comment:
I'm not sure about your math, but it would be easy to put the metadata into a separate CDS and just include some integer index in the values CDS. I couldn't get the plot labels to use the data directly from the CDS, so no real need to include it all together. Hover tools might be a little trickier but I've dealt with something similar before.